packages/cmd/src/CommandProcessor.ts

```diff
@@ -371,7 +371,9 @@ export class CommandProcessor {
             process.chdir(params.arguments.dcd as string)
 
             // reinit config for daemon client directory
-            ImperativeConfig.instance.config = await Config.load(ImperativeConfig.instance.rootCommandName, ImperativeConfig.instance.config.opts);
+            const newOpts = ImperativeConfig.instance.config?.opts || {};
+            if (newOpts.vault == null) newOpts.vault = ImperativeConfig.instance.config?.mVault;
+            ImperativeConfig.instance.config = await Config.load(ImperativeConfig.instance.rootCommandName, newOpts);
```

Comment:
This is the issue I was trying to explain in previous standups where the Daemon always prompts for credentials (even after doing `zdevx config secure --global`

Reply:
thanks - i thought i had fixed something very similar to this before, but apparently not 😄

Reply:
I thought so as well... this code seemed to fix it in the past:
https://github.com/zowe/imperative/blob/66f7b630943570faffc3016700e239584b49da6b/packages/config/src/api/ConfigSecure.ts#L32-L34
But at least you got it working 😛

Reply:
🤔 Now I'm confused.
Maybe I should remove the hack of creating the `newOpts` object?

Reply:
i think this change is needed - im pretty confused, but i started experiencing the problem @zFernand0 had.  this code change fixed the problem (the continuous need to respond to credential prompting)